tools/testsocket/testsocket.cpp

```diff
@@ -664,15 +679,55 @@ int doSendQuery(const char * ip, unsigned port, const char * base)
                 fullQuery.append(queryRemoteStreamCmd());
             fullQuery.append(base);
         }
+
+        if (generateClientSpans && clientSpan->isValid())
+        {
+            Owned<IProperties> retrievedClientHeaders = createProperties();
+            clientSpan->getClientHeaders(retrievedClientHeaders.get());
+
+            if (retrievedClientHeaders->hasProp("traceparent"))
+            {
+                //disassemble xml in oder to inject _trace header
+                Owned<IPTree> p = createPTreeFromXMLString(fullQuery.str(), ipt_none, ptr_none);
+                const char *queryName = p->queryName();
+
+                fullQuery.setf("<%s><_trace><traceparent>%s</traceparent>", queryName, retrievedClientHeaders->queryProp("traceparent"));
+
+                IPropertyTree * traceHeader = p ->queryPropTree("_trace");
```

Comment:
Would it work (and be simpler) to code as:

```
if (traceHeader)
    traceHeader->setProp("traceparent", retrievedClientHeaders->queryProp("traceparent");

```
and then convert the whole property tree to xml again?